### PR TITLE
Add YugipediaPrintMapper

### PR DESCRIPTION
**Pull request by magicDGS**
- Add YugipediaPrintMapper including a new required method in `GeneralMapper`
- Add unit-tests with extra edge-case for future consideration
- Add approval tests

### CODE REVIEW

File: tools/src/test/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/YugipediaPrintMapperTest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mapstruct.factory.Mappers;
+
+import io.github.ygojson.model.data.Print;
+import io.github.ygojson.model.utils.JsonUtils;
+import io.github.ygojson.tools.dataprovider.impl.yugipedia.YugipediaTestData;
+import io.github.ygojson.tools.dataprovider.impl.yugipedia.model.CardTable2;
+
+class YugipediaPrintMapperTest {
```
Comment (magicDGS) [inline tools/src/test/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/YugipediaPrintMapperTest.java:18]: Move to the new structure once https://github.com/ygojson/ygojson-tools/pull/60 is ready

File: tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/YugipediaPrintMapper.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+public abstract class YugipediaPrintMapper {
+
+	private static final int EXPECTED_FIELD_NUMBER = 3;
+
+	// 1st field: cardNumber - out printCode
+	private static final int CARD_NUMBER_FIELD = 0;
+	// 2nd field: setName (useful for the firstSeries cards withut printCode)
+	private static final int SET_NAME_FIELD = 1;
+	// 3rd filed: rarity list
+	private static final int RARITY_FIELD = 2;
```
Comment (magicDGS) [inline tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/YugipediaPrintMapper.java:29]: Change name to `RARITY_LIST_FIELD` to be sure that we remember that is not a single rarity.

File: tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/YugipediaPrintMapper.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+)
+public abstract class YugipediaPrintMapper {
+
+	private static final int EXPECTED_FIELD_NUMBER = 3;
+
+	// 1st field: cardNumber - out printCode
+	private static final int CARD_NUMBER_FIELD = 0;
+	// 2nd field: setName (useful for the firstSeries cards withut printCode)
+	private static final int SET_NAME_FIELD = 1;
+	// 3rd filed: rarity list
```
Comment (magicDGS) [inline tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/YugipediaPrintMapper.java:28]:
```suggestion
	// 3rd field: rarity list
```
